File: src/main/java/com/sample/taskmanager/Task.java
package com.sample.taskmanager;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.time.Duration;

public class Task {

    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");

    private final int id;
    private final String title;
    private final Priority priority;
    private Status status;
    private final LocalDateTime createdAt;
    private LocalDateTime completedAt;

    public Task(int id, String title, Priority priority) {
        this.id = id;
        this.title = title;
        this.priority = priority;
        this.status = Status.PENDING;
        this.createdAt = LocalDateTime.now();
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Priority getPriority() {
        return priority;
    }

    public Status getStatus() {
        return status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public LocalDateTime getCompletedAt() {
        return completedAt;
    }

    void markCompleted() {
        this.status = Status.COMPLETED;
        this.completedAt = LocalDateTime.now();
    }

    public boolean isOverdue(Duration threshold) {
        if (status == Status.COMPLETED) {
            return false;
        }
        return Duration.between(createdAt, LocalDateTime.now()).compareTo(threshold) > 0;
    }

    @Override
    public String toString() {
        String statusStr = (completedAt != null)
                ? "Completed at " + completedAt.format(FORMATTER)
                : status.toString();
        return String.format("[#%d] %s (Priority: %s, Status: %s)", id, title, priority, statusStr);
    }
}

File: taskmanager/task.go
package taskmanager

import (
	"fmt"
	"time"
)

// Priority represents the urgency level of a task.
type Priority int

const (
	PriorityLow    Priority = iota
	PriorityMedium
	PriorityHigh
)

// String returns the human-readable name for a Priority.
func (p Priority) String() string {
	switch p {
	case PriorityLow:
		return "Low"
	case PriorityMedium:
		return "Medium"
	case PriorityHigh:
		return "High"
	default:
		return "Unknown"
	}
}

// Valid returns true if the priority is a recognized value.
func (p Priority) Valid() bool {
	return p >= PriorityLow && p <= PriorityHigh
}

// Status represents the current state of a task.
type Status int

const (
	StatusPending   Status = iota
	StatusCompleted
)

// String returns the human-readable name for a Status.
func (s Status) String() string {
	switch s {
	case StatusPending:
		return "Pending"
	case StatusCompleted:
		return "Completed"
	default:
		return "Unknown"
	}
}

// Task represents a single to-do item.
type Task struct {
	ID          int
	Title       string
	Priority    Priority
	Status      Status
	CreatedAt   time.Time
	CompletedAt *time.Time
}

// String returns a formatted representation of the task.
func (t Task) String() string {
	status := t.Status.String()
	if t.CompletedAt != nil {
		status = fmt.Sprintf("Completed at %s", t.CompletedAt.Format(time.RFC822))
	}
	return fmt.Sprintf("[#%d] %s (Priority: %s, Status: %s)", t.ID, t.Title, t.Priority, status)
}

// IsOverdue checks if a pending task was created more than the given duration ago.
func (t Task) IsOverdue(threshold time.Duration) bool {
	if t.Status == StatusCompleted {
		return false
	}
	return time.Since(t.CreatedAt) > threshold
}
